### DISCUSSION TITLE
什么时候能支持 exports 字段导出的 npm包呢？？？

### DISCUSSION THREAD

**Discussion by wzy2023** [umijs/umi#11408] [Q&A]
package.json 里用了 exports 情景导出的包，什么时候可以支持呢？

**Comment by fz6m** [comment]
如在 umi 3 中遇到 esm / exports / mjs 相关的包报错问题，可以参考 https://github.com/umijs/umi/issues/7935#issuecomment-1074783801 解决。

如在 umi 4 中如遇到问题，需要给一个最小复现帮助排查问题。

**Comment by wzy2023** [comment]
@fz6m 你好，codeStandbox: https://codesandbox.io/s/eloquent-diffie-vdftc8?file=/.umirc.ts

使用了自己写的 @wzyjs/utils 包，这个包用了 exports 字段情景导出

默认方式启动的话，可以成功
但使用 vite 启动时，会报错

**Reply by fz6m** [reply]
没遇到你说的这个问题，如有问题，给一个 github repo 最小复现把，我试了下是可以的。

**Reply by wzy2023** [reply]
诶 看这个 codeStandbox 不行吗 
https://codesandbox.io/s/eloquent-diffie-vdftc8?file=/src/pages/Home/index.tsx

不行的话，我稍后建个 github repo 出来

**Reply by fz6m** [reply]
这里面是 `4.0.13` 旧版本的，用最新版本的试试，我在本地试了是没问题的。如还有问题，给一个 github repo 链接吧。

**Reply by wzy2023** [reply]
还是有问题哦 
node 18.10.0
pnpm 7.30.5

git@github.com:wzy2023/test-umi.git

**Reply by fz6m** [reply]
需要写成：

```ts
import * as utils from '@wzyjs/utils'
```

你这个包的产物没有互操作的胶水和提示代码，导致 vite 不能分析到这种具名导入，而 webpack 对此进行了胶水层兼容。

自己排查下构建工具的问题吧，更推荐使用 [father](https://github.com/umijs/father) 打包。

**Reply by wzy2023** [reply]
感谢大佬，这样写确实有用！

另外想请教下，我现在用的是 rollup，有什么插件可以支持产出【互操作的胶水和提示代码】吗

**Reply by wzy2023** [reply]
不太懂【互操作的胶水和提示代码】是什么，

可以给个关键词吗，我去搜索学习一下~~~

**Comment by wzy2023** [comment]
@fz6m 大佬求帮助 我周六日问了好几个人，他们都不知道胶水代码是什么。。

**Reply by wzy2023** [reply]
@fz6m 用father打包也是一样的 报错
